x/mongo/driver/topology/topology_test.go

```diff
@@ -693,6 +693,217 @@ func TestTopologyConstruction(t *testing.T) {
 	})
 }
 
+type mockLogSink struct {
+	msgs []string
+}
+
+func (s *mockLogSink) Info(_ int, msg string, _ ...interface{}) {
+	s.msgs = append(s.msgs, msg)
+}
+func (*mockLogSink) Error(error, string, ...interface{}) {
+	// Do nothing.
+}
+
+// Note: SRV connection strings are intentionally untested, since initial
+// lookup responses cannot be easily mocked.
+func TestTopologyConstructionLogging(t *testing.T) {
+	sink := &mockLogSink{}
+	loggerOptions := options.
+		Logger().
+		SetSink(sink).
+		SetComponentLevel(options.LogComponentTopology, options.LogLevelInfo)
+	t.Run("CosmosDB URIs", func(t *testing.T) {
+		testCases := []struct {
+			name string
+			uri  string
+			msgs []string
+		}{
+			{
+				name: "normal",
+				uri:  "mongodb://a.mongo.cosmos.azure.com:19555/",
+				msgs: []string{cosmosDBMsg},
```

Comment:
We should directly reference the cosmosDBMsg in the test, and not use a non-test constant. There are a few reasons for this:

1. Test code won't change when production code is refactored. That is, if we use a constant here then refactoring the constant in the production code would break this test.
2. These tests might pass if the code is wrong. Any changes to the constants will be considered valid, for instance if we accidentally append a character to the constant.

This applies to the other test cases as well.